core/src/main/scala/cats/syntax/functor.scala

```diff
@@ -4,6 +4,9 @@ package syntax
 trait FunctorSyntax extends Functor.ToFunctorOps {
   implicit final def catsSyntaxFunctorTuple2Ops[F[_], A, B](fab: F[(A, B)]): FunctorTuple2Ops[F, A, B] =
     new FunctorTuple2Ops[F, A, B](fab)
+
+  implicit final def catsSyntaxIfF[F[_]: Functor](fa: F[Boolean]): IfFOps[F] =
+    new IfFOps[F](fa)
```

Comment:
Remove `: Functor` constraint from this method please: it is passed to the `ifF` method itself, not the `IfOps` class constructor. Therefore for this method it is completely redundant.
```suggestion
  implicit final def catsSyntaxIfF[F[_]](fa: F[Boolean]): IfFOps[F] =
    new IfFOps[F](fa)
```

Reply:
I'm thinking... Wouldn't it be better to rename `IfFOps` and `catsSyntaxIfF` into something like `FunctorBooleanOps` and `catsSyntaxFunctorBooleanOps`? Seems it would closer follow the naming convention used in Cats (see the above `FunctorTuple2Ops` as an example).

@armanbilge @atais wdyt?

Reply:
I wondered the same thing, but this follows the naming convention used by `ifM`.

Reply:
Whatever... I'm ok with both options :)

Reply:
I would say `FunctorBooleanOps` would look more typical to me since potentially the class can cover all syntax for `F[Boolean]`, not only `ifF` method. But on the other hand, there's no other method in `Functor` that requires `F[Boolean]` besides `ifF`. And I wonder if it is even possible to come up with another one. So at this moment both names `IfFOps` and `FunctorBooleanOps` look pretty legit.

Reply:
Actually, I'm not sure we want to make this change. Esp. if `ifM` is implemented in the same way. The constraint may help with implicit search for syntax.

Reply:
Yeah, now I am not sure either. It seems there's a lack of consistency among such methods is Cats in general, unfortunately. Some of the methods do require an typeclass constraint (even though it is not used) but some others do not:

* `MonadSyntax#catsSyntaxMonad` do not require `Monad`
* `AlternativeSyntax` both methods do not require `Alternative`
* `ApplicativeSyntax` do not require BUT
* `ApplicativeErrorSyntax#catsSyntaxApplicativeError` DO require `ApplicativeError` along with
* `FlatMapSyntax` and `FoldableSyntax` methods which also DO require their corresponding constraints.
* and so on – I haven't checked everything )

I guess it depends mostly on a contributor's personal preference rather than on a specific convention.

Reply:
I raised a question regarding the convention [here](https://discord.com/channels/632277896739946517/633329569402978313/951211557231140894).

Reply:
soo... do guys want to do anything about this issue in this PR?
if not, btw, I think I cannot merge myself.

Reply:
Well, as @Daenyth suggested in the Discord channel, the difference will affect compiler error messages. Check this out:
```scala
import cats.syntax.all._

case class Foo[A](a: A)

Foo(true).ifM(1, 2)
```
This snippet won't compile since there's no `Functor[Foo]` defined apparently. But if we leave the implicit method with the `Functor` constraint:
```
implicit final def catsSyntaxIfF[F[_]: Functor](fa: F[Boolean]): IfFOps[F] = ???
```
then the error message will be the following:
```
Foo(true).ifF(1, 2)
          ^
error: value ifF is not a member of Foo[Boolean]
```
but if we remove the constraint in the way I suggested initially:
```
implicit final def catsSyntaxIfF[F[_]](fa: F[Boolean]): IfFOps[F] = ???
```
then the error will look like
```
Foo(true).ifF(1, 2)
             ^
error: could not find implicit value for parameter F: cats.Functor[Foo]
```
Therefore we can justify whether to leave or remove the constraint basing on which error message looks clearer and easier to comprehend:)

Reply:
@atais @armanbilge what would you prefer?

My personal preference: I would say the latter looks better for me. But I'm not going to insist on it – feel free to merge if you think otherwise.

Reply:
I like the second one more too.

Let me fix that.

Sorry i am not the member of the discord channel and in such situation the link does not open the discussion.

What happened to Gitter, is it not good anymore ;)? At least it was indexed in Google :)

Reply:
squashed my changes and pushed. thanks guys!

Reply:
Well, there was some discussion some time ago and a decision to step back from Gitter in favor of Discord eventually. I wasn't following all the details but yeah Discord is a better option for now.